sklego/loess.py

```diff
@@ -0,0 +1,332 @@
+            self.indices[index] = x_indices
+        else:  # If the number of returned indices is too small, resort to knn_symmetric
+            self._get_knn_symmetric_indices(index, x_focal, x, n_points)
+
+    def _get_knn_window_indices(self, index, x_focal, x, n_points):
+        x_focal = np.asarray([x_focal]).reshape(-1, 1)
+
+        knn = NearestNeighbors(n_neighbors=int(n_points)).fit(x.reshape(-1, 1))
+        self.indices[index] = knn.kneighbors(x_focal)[1][0]
+
+    def _get_knn_symmetric_indices(self, index, x_focal, x, n_points):
+
+        low_mask = x < x_focal
+        high_mask = ~low_mask
+
+        x_low = x[low_mask]
+        x_high = x[high_mask]
+
+        x_focal = np.asarray([x_focal]).reshape(-1, 1)
+
+        x_low_indices = np.where(low_mask)[0]
+        x_high_indices = np.where(high_mask)[0]
+
+        if (x_low.size > floor(n_points / 2)) & (x_high.size > floor(n_points / 2)):
+            knn_low = NearestNeighbors(n_neighbors=int(n_points / 2)).fit(x_low.reshape(-1, 1))
+            knn_high = NearestNeighbors(n_neighbors=int(n_points / 2)).fit(x_high.reshape(-1, 1))
+            indices_low_local = knn_low.kneighbors(x_focal)[1][0]
+            indices_high_local = knn_high.kneighbors(x_focal)[1][0]
+
+            indices_low = x_low_indices[indices_low_local]
+            indices_high = x_high_indices[indices_high_local]
+            self.indices[index] = np.concatenate([indices_low, indices_high])
+
+        elif (x_low.size > floor(n_points / 2)) & (not x_high.size > floor(n_points / 2)):
+            knn_low = NearestNeighbors(n_neighbors=int(n_points)).fit(x_low.reshape(-1, 1))
+            indices_low_local = knn_low.kneighbors(x_focal)[1][0]
+            self.indices[index] = x_low_indices[indices_low_local]
+
+        else:
+            knn_high = NearestNeighbors(n_neighbors=int(n_points)).fit(x_high.reshape(-1, 1))
+            indices_high_local = knn_high.kneighbors(x_focal)[1][0]
+            self.indices[index] = x_high_indices[indices_high_local]
+
+    def _fit_model_per_window(self, x, y):
+        """
+        With the created x focal points and accompanying windows, fit per x focal point a model on
+        the local window and predict an y focal using the fitted model.
+        :param x:
+        :param y:
+        """
+        for index in self.indices.keys():
+            x_focal = np.asarray([self.x_focal_base[index]]).reshape(-1, 1)
+            x_window = x[self.indices[index]].reshape(-1, 1)
+            y_window = y[self.indices[index]].reshape(-1, 1)
+
+            # TODO: implement iterative weighted fit
+            model = self.model.fit(x_window, y_window)
+            self.y_focal_base = np.append(self.y_focal_base, model.predict(x_focal))
+            self.model_per_window = np.append(self.model_per_window, model)
+
+
+def plot_windows(x, y, x_focal_base, y_focal_base, indices):
+    """
+    TODO: Use matplotlib.animation to create video
```

Comment:
if we want to make this animation availble it might be better to place this somewhere in the documentation. perhaps add a script to a new `/examples` folder in root?